server/src/main/java/org/elasticsearch/cluster/routing/allocation/allocator/DesiredBalanceReconciler.java

```diff
@@ -504,10 +520,14 @@ private void balance() {
                 }
             }
 
-            maybeLogUndesiredAllocationsWarning(allAllocations, undesiredAllocations, routingNodes.size());
+            DesiredBalanceReconciler.this.unassignedShards.set(unassignedShards);
+            DesiredBalanceReconciler.this.undesiredAllocations.set(undesiredAllocations);
+            DesiredBalanceReconciler.this.totalAllocations.set(totalAllocations);
```

Comment:
While APM integration is not completed yet, we might need to log those periodically to set up dashboards

Reply:
I see the other existing metrics in DesiredBalanceStats use the `CounterMetric` class, which I assume is the old way of stats.
And I see for APM there's the `LongCounter` class (see e.g., https://github.com/elastic/elasticsearch/pull/100383).
And you used an AtomicInteger directly in this PR.

I am a bit confused about which one of the above should be used. I wonder, if APM integration is the ultimate aim, whether it would make sense to use a LongCounter in this PR? To avoid later having a duplicate variable for APM.

Reply:
Please note this is setting value, not incrementing, so this can not be LongCounter/CounterMetric, but should be some sort of Gauge instead.
Additionally LongGauge/LongCounter do not expose their current state. So we would need a combination of LongGauge+AtomicLong or LongGauge+AtomicLong to have apm integration and be able to provide stats to the rest endpoint at the same time (unless they cahnge the signature of those interfaces).

I am planning to integrate LongCounter/LongGauge for all existing allocator metrics as a separate PR

Reply:
Ah if they do not expose their current state, then OK.